pkg/controller/registry/reconciler/reconciler.go

```diff
@@ -195,19 +192,29 @@ func Pod(source *operatorsv1alpha1.CatalogSource, name string, image string, saN
 			NodeSelector: map[string]string{
 				"kubernetes.io/os": "linux",
 			},
-			SecurityContext: &corev1.PodSecurityContext{
-				SeccompProfile: &corev1.SeccompProfile{
-					Type: corev1.SeccompProfileTypeRuntimeDefault,
-				},
-			},
 			ServiceAccountName: saName,
 		},
 	}
 
-	if runAsUser > 0 {
-		pod.Spec.SecurityContext.RunAsUser = &runAsUser
-		pod.Spec.SecurityContext.RunAsNonRoot = pointer.Bool(true)
+	if source.Spec.GrpcPodConfig != nil && source.Spec.GrpcPodConfig.RequirePrivilegedAccess {
+		pod.Spec.Containers[0].SecurityContext.AllowPrivilegeEscalation = pointer.Bool(true)
```

Comment:
@perdasilva i think you had done the deepest investigation on what permissions our legacy catalogs/opm binary required to be able to run on a 4.12 cluster (in the process of you fixing the issues so we didn't require those permissions anymore), can you take a look at whether this seems right?  It's not what i would have expected.

Reply:
Here's what it came down to:
```
securityContext:
      allowPrivilegeEscalation: false
      capabilities:
        drop:
        - MKNOD
      readOnlyRootFilesystem: false
```
and
```
securityContext:
    runAsNonRoot: false
    runAsUser: 0
    seccompProfile:
      type: RuntimeDefault
```

This ran on a `baseline` enforced namespace. Here's something curious though, @bparees, I only managed to change the PSA enforcement profile on a (new) `openshit-` namespace. I.e.

```
$ oc create ns test
$ oc describe ns test
...
Labels:  kubernetes.io/metadata.name=test
              pod-security.kubernetes.io/enforce=restricted
              pod-security.kubernetes.io/enforce-version=v1.24
...
$ oc label ns test pod-security.kubernetes.io/enforce=baseline --overwrite
$ oc describe ns test
...
Labels:  kubernetes.io/metadata.name=test
              pod-security.kubernetes.io/enforce=restricted
              pod-security.kubernetes.io/enforce-version=v1.24
...
```

So, I think we may still have problems even with the api override =S

Reply:
Fyi @perdasilva not sure if you saw [this conversation](https://github.com/operator-framework/operator-lifecycle-manager/pull/2845#discussion_r957031708), but it might help fill in the context for why I switched from setting `runAsUser: 0` to not setting it at all. Given the fact that we're saying to run legacy catalogs, admins need to make their namespaces look like what it was pre PSA restricted world, it does make sense to make the registry pods look like what they were pre PSA restricted world too, and in that world we did not set `runAsUser:0`, unless there are specific reasons why we should be setting `runAsUser:0` now.

Reply:
> So, I think we may still have problems even with the api override =S

@perdasilva i believe you need to explicitly label the NS as having the labelsyncer disabled, before you can set the PSA labels manually.  It should be the same label you are setting on openshift-* NSes to enable the labelsyncer there (except you'd set it to false, of course, here)

non payload openshift-* NSes default to labelsyncer on (so you need to label the NS appropriately to disable it)
openshift-* NSes default to labelsyncer off (so you need to label the NS appropriately to enable it)

Reply:
doh! should have inferred that :facepalm:

Reply:
> Fyi @perdasilva not sure if you saw [this conversation](https://github.com/operator-framework/operator-lifecycle-manager/pull/2845#discussion_r957031708), but it might help fill in the context for why I switched from setting `runAsUser: 0` to not setting it at all. Given the fact that we're saying to run legacy catalogs, admins need to make their namespaces look like what they were pre PSA restricted world, it does make sense to make the registry pods look like what they were pre PSA restricted world too, and in that world we did not set `runAsUser:0`, unless there are specific reasons why we should be setting `runAsUser:0` now.

So, not setting `runAsUser` seemed to push the restricted-v2 scc profile causing the perm issue. If we don't want to set it to 0, we'll probably need to set the `fsGroup` to 0 in lieu of that. I've also tested by removing runAsUser and runAsNonRoot and setting AllowPriviledgeEscalation to true and that worked as well. Maybe that's the best way forward, then.

Reply:
> So, not setting runAsUser seemed to push the restricted-v2 scc profile causing the perm issue. If we don't want to set it to 0, we'll probably need to set the fsGroup to 0 in lieu of that. 

if you do not set the seccompprofile, and you have access to the anyuid SCC, i believe you'll get the anyuid SCC which allows you to leave the runAsUser unset, and the container will run w/ the image's default UID, which i believe is what you want.